src/osa/scripts/gainsel_webmaker.py

```diff
@@ -0,0 +1,139 @@
+    gainsel_df['GainSelStatus'] = np.where(gainsel_df['failed'] != 0,
+                                           'FAILED',
+                                           np.where(gainsel_df['pending'] != 0,
+                                                    'PENDING',
+                                                    'COMPLETED'))
+
+    total_job_number = gainsel_df['pending'] + gainsel_df['failed'] + gainsel_df['success']
+    gainsel_df['GainSel%'] = round(gainsel_df['success'] * 100 / total_job_number, 1)
+
+    summary_table = summary_table.to_pandas()
+
+    final_table = pd.merge(summary_table, gainsel_df, on="run_id")[['run_id',
+                                                                    'n_subruns',
+                                                                    'run_type',
+                                                                    'pending',
+                                                                    'success',
+                                                                    'failed',
+                                                                    'GainSelStatus',
+                                                                    'GainSel%']]
+
+    return final_table
+
+def main():
+    """Produce the html file with the processing OSA Gain Selection status."""
+    args = ArgumentParser(
+        description="Script to make an xhtml from LSTOSA sequencer output", parents=[common_parser]
+    ).parse_args()
+
+    if args.date:
+        flat_date = date_to_dir(args.date)
+        options.date = args.date
+
+    else:
+    # yesterday by default
+        yesterday = datetime.now() - timedelta(days=1)
+        options.date = yesterday
+        flat_date = date_to_dir(yesterday)
+
+    date = date_to_iso(options.date)
+    run_summary_directory = Path(cfg.get("LST1", "RUN_SUMMARY_DIR"))
+    run_summary_file = run_summary_directory / f"RunSummary_{flat_date}.ecsv"
+
+    if not run_summary_file.is_file() or len(Table.read(run_summary_file)["run_id"]) == 0:
+
+        html_table = "<p>No data found</p>"
+        # Save the HTML file
+        directory = Path(cfg.get("LST1", "GAIN_SELECTION_WEB_DIR"))
+        directory.mkdir(parents=True, exist_ok=True)
+        html_file = directory / f"osa_gainsel_status_{flat_date}.html"
+        html_file.write_text(html_content(html_table, date, "OSA Gain Selection"), encoding="utf-8")
+
+    else:
+       # Get the table with the gain selection check report:
+        table_gain_selection_jobs = check_failed_jobs(datetime.strptime(date, "%Y-%m-%d"))
+
+        table_gain_selection_jobs.reset_index(drop=True, inplace=True)
+        html_table = table_gain_selection_jobs.to_html()
+
+        # Save the HTML file
+        directory = Path(cfg.get("LST1", "GAIN_SELECTION_WEB_DIR"))
+
+        directory.mkdir(parents=True, exist_ok=True)
+
+        html_file =open( directory / f"osa_gainsel_status_{date}.html", "w")
```

Comment:
```suggestion
        html_file =open( directory / f"osa_gainsel_status_{flat_date}.html", "w")
```
To be consistent with the other if clause